src/main/kotlin/org/rust/ide/annotator/RustHighlightingAnnotator.kt

```diff
@@ -31,6 +35,15 @@ class RustHighlightingAnnotator : Annotator {
 
         override fun visitTypeParam(o: RustTypeParamElement) {
             holder.highlight(o, RustColor.TYPE_PARAMETER)
+            o.typeParamBounds?.let {
+                holder.highlight(it.colon, RustColor.KEYWORD) // feels off
+                it.polyboundList.forEach {
+                    it.bound.traitRef?.let {
+                        // Should this visit path instead?
+                        holder.highlight(it.path.identifier, RustColor.TRAIT)
+                    }
+                }
+            }
```

Comment:
Anyway, highlighting trait reference should be moved to dedicated visitor method.